librz/analysis/esil/esil_trace.c

```diff
@@ -298,132 +335,44 @@ RZ_API void rz_analysis_esil_trace_restore(RzAnalysisEsil *esil, int idx) {
 	ht_up_foreach(trace->memory, restore_memory_cb, esil);
 }
 
-static int cmp_strings_by_leading_number(void *data1, void *data2) {
-	const char *a = sdbkv_key((const SdbKv *)data1);
-	const char *b = sdbkv_key((const SdbKv *)data2);
-	int i = 0;
-	int j = 0;
-	int k = 0;
-	while (a[i] >= '0' && a[i] <= '9') {
-		i++;
-	}
-	while (b[j] >= '0' && b[j] <= '9') {
-		j++;
-	}
-	if (!i) {
-		return 1;
-	}
-	if (!j) {
-		return -1;
-	}
-	i--;
-	j--;
-	if (i > j) {
-		return 1;
-	}
-	if (j > i) {
-		return -1;
-	}
-	while (k <= i) {
-		if (a[k] < b[k]) {
-			return -1;
-		}
-		if (a[k] > b[k]) {
-			return 1;
-		}
-		k++;
-	}
-	for (; a[i] && b[i]; i++) {
-		if (a[i] > b[i]) {
-			return 1;
-		}
-		if (a[i] < b[i]) {
-			return -1;
-		}
-	}
-	if (!a[i] && b[i]) {
-		return -1;
-	}
-	if (!b[i] && a[i]) {
-		return 1;
-	}
-	return 0;
+static void print_instruction_trace(RzILTraceInstruction *instruction, int idx) {
+	printf("[%d]instruction addr -> %lld\n", idx, instruction->addr);
 }
 
+/**
+ * List all traces
+ * \param esil RzAnalysisEsil *, ESIL instance
+ */
 RZ_API void rz_analysis_esil_trace_list(RzAnalysisEsil *esil) {
 	if (!esil->trace) {
 		return;
 	}
 
-	PrintfCallback p = esil->analysis->cb_printf;
-	SdbKv *kv;
-	SdbListIter *iter;
-	SdbList *list = sdb_foreach_list(esil->trace->db, true);
-	ls_sort(list, (SdbListComparator)cmp_strings_by_leading_number);
-	ls_foreach (list, iter, kv) {
-		p("%s=%s\n", sdbkv_key(kv), sdbkv_value(kv));
+	RzILTraceInstruction *instruction_trace;
+	int idx = 0;
+	void **iter;
+	rz_pvector_foreach (esil->trace->instructions, iter) {
+		instruction_trace = *iter;
+		print_instruction_trace(instruction_trace, idx);
+		idx++;
 	}
-	ls_free(list);
 }
 
+/**
+ * Display an ESIL trace at index `idx`
+ * \param esil RzAnalysisEsil *, ESIL instance
+ * \param idx int, index of trace
+ */
 RZ_API void rz_analysis_esil_trace_show(RzAnalysisEsil *esil, int idx) {
+	printf("Trace Show : WIP\n");
```

Comment:
Was this forgotten?

Reply:
I plan to add this in other pr. Because when I want to figure out the proper output of it, there is no info about {dte} things. So it will be better to leave an WIP and add test about it in the future.

Reply:
I'm trying to run your PR but without anything printed when doing `dte <idx>` it's hard to check if anything is working well at all.. I think it would be great to have at least a raw dump of all fields within the esiltrace instruction.

Reply:
Yeah that's weird, during the time I'm exploring the old esil trace I found {dte} command families print nothing. I rebuild the `dev` branch today and reproduce it. All trace logs seems be freed after analysis done.

Reply:
Yes, I confirm that even with those commands `dte` prints nothing. It seems either unimplemented or a bug, should not be a part of this PR.
```
aeim
aei
aets+
aes
aes
# check the registers to see if stepped succesfully
dte
```

Reply:
Moreover, if I enable assertions, `aets+` prints the following error:
```
[0x00006b10]> aets+
WARNING: rz_analysis_esil_trace_new: assertion 'esil && esil->stack_addr && esil->stack_size' failed (line 22)
```

Reply:
I made a new issue to track this: https://github.com/rizinorg/rizin/issues/1416